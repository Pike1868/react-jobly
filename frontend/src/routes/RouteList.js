import React from "react";
import { Routes, Route, Navigate } from "react-router-dom";
import HomePage from "../pages/HomePage";
import SignIn from "../pages/SignIn";
import SignUp from "../pages/SignUp";
import { useUserContext } from "../context/UserContext";
import CompaniesPage from "../pages/CompaniesPage";
import CompanyProfile from "../pages/CompanyProfile";
import JobsPage from "../pages/JobsPage";
import UserProfile from "../pages/UserProfile";

/**
 * RouteList Component:
 *
 * Defines all routes for app
 *  ---Available when logged in:
 * "/": Displays homepage and user welcome message
 * "/companies": Displays CompanyPage with all companies
 * "/jobs": Displays JobsPage with all jobs
 * "/profile": Displays UserProfile with user info
 * "/companies/:handle": Displays CompanyProfile page with all jobs for that company
 *
 * "/signin": Displays form for existing users to sign in
 * "/signup": Displays form for new users to sign up
 * - "/*": A catch-all route that redirects any paths with no matches
 *
 */

// RouteList Component
const RouteList = () => {
  const { user } = useUserContext();
  return (
    <Routes>
      {user ? (
        <>
          <Route path="/" element={<HomePage />} />
          <Route path="/companies" element={<CompaniesPage />} />
          <Route path="/jobs" element={<JobsPage />} />
          <Route path="/profile" element={<UserProfile />} />
          <Route path="/companies/:handle" element={<CompanyProfile />} />
          <Route path="/*" element={<Navigate to="/" />} />
        </>
      ) : (
        <>
          <Route path="/signin" element={<SignIn />} />
          <Route path="/signup" element={<SignUp />} />
          <Route path="/*" element={<Navigate to="/signin" />} />
        </>
      )}
    </Routes>
  );
};

export default RouteList;
