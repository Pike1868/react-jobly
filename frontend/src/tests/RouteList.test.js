import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import RouteList from "../routes/RouteList";
import { mockUserDetails } from "./commonSetup";

jest.mock("../context/UserContext", () => ({
  useUserContext: () => mockUserDetails,
}));

describe("RouteList", () => {
  it("renders without crashing", () => {
    render(
      <MemoryRouter>
        <RouteList />
      </MemoryRouter>
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <RouteList />
      </MemoryRouter>
    );

    expect(asFragment).toMatchSnapshot();
  });

  it("renders HomePage when user is logged in", () => {
    render(
      <MemoryRouter>
        <RouteList />
      </MemoryRouter>
    );

    const element = screen.getByText(/Welcome Back/);

    expect(element).toBeInTheDocument();
  });
});
