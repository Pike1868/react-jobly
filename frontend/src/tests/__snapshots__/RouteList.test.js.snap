// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteList matches snapshot 1`] = `[Function]`;
